### PR TITLE
add the option for specifying the ssh-proxy port /  improve python2/3 compatibility

### PR DESCRIPTION
**Pull request by r-xue**

### CODE REVIEW

File: adsbibdesk.py [hunk 1]
```diff
@@ -638,20 +657,21 @@ def has_annotationss(f):
...
+    #                                          'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36'})
+    #     else:
+    #         response = requests.get(url)
+    #response.raise_for_status()
+    #return response.url
+    try:
+        out = urlopen(url)
+    except URLError as out:
+        return out.geturl()
+    return out.geturl()
```
[the remark this answers is not in the document]
Comment (r-xue) [reply adsbibdesk.py]: hold on. let me take a look. I tested it under python 2.7 and 3.7 on local and campus machines, but only for ApJ/arXiv

File: adsbibdesk.py [hunk 2]
```diff
@@ -1152,7 +1174,8 @@ def handle_starttag(self, tag, attrs):
             # abstract
             try:
                 self.abstract = self.tag.strip().decode('utf-8')
-            except AttributeError:
+            #except AttributeError:
+            except Exception:
```
[the remark this answers is not in the document]
Comment (r-xue) [reply adsbibdesk.py]: I got this under 2.7; 3.7 is fine.
```
UnicodeEncodeError: 'ascii' codec can't encode character u'\u02dc' in position 537
```

File: adsbibdesk.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (r-xue) [reply adsbibdesk.py]: well... looks like we still need the "browsers" workaround. I will fix it.

File: adsbibdesk.py [hunk 3]
```diff
@@ -1330,9 +1330,12 @@ def filetype(filename):
                         is not None:
                     fd, pdf = tempfile.mkstemp(suffix='.pdf')
                     cmd = 'ssh -p %s %s@%s \"touch /tmp/adsbibdesk.pdf; ' \
-                        'wget -O /tmp/adsbibdesk.pdf \\"%s\\"\"' % \
+                        'wget -O /tmp/adsbibdesk.pdf ' \
+                        '--header=\\"Accept: text/html\\" ' \
+                        '--user-agent=\\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36\\" ' \
+                        '\\"%s\\"\"' % \
```
Comment (r-xue) [inline adsbibdesk.py]: For MNRAS articles, the proxy method failed, because the redirected link seen from a machine outside the paywall doesn't contain any PDF information. Here we will just let wget on the proxy follow the ads links directly.

**Comment by r-xue** [comment]
The code seems to be work for AAS journals and MNRAS now ( via both the local or proxy method). I am gradually moving my literature collections from Papers3 to BibDesk. If I notice something else broken, I will bring it up to you or try to patch it directly.

File: adsbibdesk.py [hunk 4]
```diff
@@ -657,6 +658,16 @@ def has_annotationss(f):
 
 def get_redirect(url):
     """Utility function to intercept final URL of HTTP redirection"""
+    """
+        EJOURANL:
```
Comment (keflavich) [inline adsbibdesk.py]: typo

File: adsbibdesk.py [hunk 5]
```diff
@@ -1284,28 +1307,28 @@ def filetype(filename):
                 print("Could not download from URL {0} because of error {1}"
                       .format(url, ex))
 
-            if resolved_url is not None:
+            if  resolved_url is not None:
```
Comment (keflavich) [inline adsbibdesk.py]: unnecessary whitespace here

File: adsbibdesk.py [hunk 6]
```diff
@@ -1284,28 +1307,28 @@ def filetype(filename):
...
+                
                 logging.debug("Resolve article URL: %s" % resolved_url)
                 if "filetype=.pdf" in resolved_url:
                     # URL will directly resolve into a PDF
                     pdf_url = resolved_url
-                elif "MNRAS" in resolved_url:
-                    # Special case for MNRAS URLs to deal with iframe
+                elif "EJOURNAL" in url:
+                    # Special case for EJOURANL URLs
```
Comment (keflavich) [inline adsbibdesk.py]: typo

**Comment by keflavich** [comment]
Is this ready to merge?   I wish we had a test suite... I have no idea whether this is going to break anything for me... but thanks for doing the tests and contributing this!

**Comment by keflavich** [comment]
What changes did you make there?  Might they be generally useful?  I've inherited maintenance responsibility for this project, so there's a lot about the code base I don't understand fully yet.

**Comment by r-xue** [comment]
from my understanding (see [CONTRIBUTING.md](https://github.com/jonathansick/ads_bibdesk/blob/master/CONTRIBUTING.md)), just need to run: 
```
python setup.py install
```

**Comment by keflavich** [comment]
Oh, right, you're saying the workflow changed when you ran the install script, but you didn't commit that?  Sounds fine.

**Comment by keflavich** [comment]
Alright, I'm merging this, but I may have to revert if I discover bugs...

**Comment by r-xue** [comment]
sounds good!
